odxtools/endofpdufield.py

```diff
@@ -156,8 +156,8 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
     def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
         """Recursively resolve any short-name references"""
         if self.structure_snref is not None:
-            dops = diag_layer.diag_data_dictionary_spec.data_object_props
-            self._structure = dops[self.structure_snref]
+            structures = diag_layer.diag_data_dictionary_spec.structures
+            self._structure = structures[self.structure_snref]
 
         if self.env_data_desc_snref is not None:
             dops = diag_layer.diag_data_dictionary_spec.data_object_props
```

Comment:
here, the same mistake happens with DOPs instead of the environment data descriptions. can you fix it as well?